### ISSUE TITLE
NeuralODE not training with DiffEqFlux v1.1.1

### ISSUE THREAD

**Issue by ali-ramadhan** [JuliaDiffEq/DiffEqFlux.jl#127]
I'm upgrading my neural ODE setups to use DiffEqFlux v1.1.1 (updating from `neural_ode` to `NeuralODE`) but I noticed that my neural ODEs don't train anymore, i.e. `Flux.train!` does not change the weights of the neural network.

I'm probably just doing something dumb but I've tried playing around with changing the arguments to `NeuralODE` and changing loss functions but no luck.

Julia environment (irrelevant packages omitted):
```
(v1.3) pkg> st
    Status `~/.julia/environments/v1.3/Project.toml`
  [aae7a2af] DiffEqFlux v1.1.1
  [0c46a032] DifferentialEquations v6.10.1
  [587475ba] Flux v0.10.1
```

Minimal working example:
```julia
using DifferentialEquations
using DiffEqFlux
using Flux

n = 10  # number of ODEs
tspan = (0.0, 1.0)

d = 5  # number of data pairs
x_data = rand(n, 5)
y_data = rand(n, 5)
training_data = [(x_data[:, i], y_data[:, i]) for i in 1:d]

NN = Chain(Dense(n, 10n, tanh),
           Dense(10n, n))

nODE = NeuralODE(NN, tspan, ROCK4(), reltol=1e-4, saveat=[tspan[end]])

loss_function(x, y) = Flux.mse(y, nODE(x))

function cb()
    train_loss = sum([loss_function(training_data[i]...) for i in 1:d])
    println("train_loss = $train_loss")
    return
end

Flux.train!(loss_function, Flux.params(NN), training_data, ADAM(1e-3), cb=cb)
```

Output:
```
train_loss = 13.049711264309027
train_loss = 13.049711264309027
train_loss = 13.049711264309027
train_loss = 13.049711264309027
train_loss = 13.049711264309027
```

**Comment by ChrisRackauckas** [comment]
`Flux.params(nODE)` since the neural differential equation parameters can be a different reference. FWIW, the newer layer functions get rid of this global nonsense.

**Comment by ali-ramadhan** [comment]
Ah I did not realize this had changed, that indeed fixed it. Thanks for taking a look and catching my mistake!

Looks like a good time to switch to `sciml_train!`.
